### PR TITLE
Publish Joint forces 

### LINKED ISSUE #883: JointStatePublisher does not publish efforts
## Environment
* OS Version: Ubuntu 20.04 
* Source or binary build?
Edifice Binary / ROS2 Foxy

Tested using both ODE and DART physics in the world file.

## Description
* Expected behavior: Joint States Publisher publishing efforts.
* Actual behavior:  Joint States Publisher (ignition::gazebo::systems::JointStatePublisher) does not publish efforts. Position and velocities look fine.

## Steps to reproduce
1. Launch any robot in ignition with ignition::gazebo::systems::JointStatePublisher plugin.
2. Start ROS2 parameter bridge
3. Echo /joint_states

Appreciate any insights / thoughts.

### PR DESCRIPTION
**Pull request by Fixit-Davide**
# 🦟 Bug fix

Fixes #883 

## Summary
These changes should allow to obtain information regarding the forces applied to the joints.

## Test it
- Pull gz-physics PR: https://github.com/gazebosim/gz-physics/pull/628
- Launch `gz sim --physics-engine gz-physics-bullet-featherstone-plugin lift_drag.sdf` simulation.
- Send a command to the model (`gz topic -t "/model/lift_drag_demo_model/joint/rod_1_joint/cmd_force" -m gz.msgs.Double  -p "data: 0.8"`).
- Echo the topic: `gz topic -e -t /world/lift_drag/model/lift_drag_demo_model/joint_state`
## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
do  you mind to sign the commit ?

**Comment by azeey** [comment]
Also, can you please add a test?

**Comment by Fixit-Davide** [comment]
I have just made a PR in gz-physics in order to populate the force fields while using bullet-featherstone. I added some information on how to test the code there (I have reported them here also).

**Review by iche033 — DISMISSED** [review]
looks good to me

**Review by iche033 — COMMENTED** [review]
> Also, can you please add a test?

oh just saw that a test was requested.

File: src/systems/physics/Physics.cc [hunk 1]
```diff
@@ -116,10 +116,13 @@
 #include "gz/sim/components/JointPositionLimitsCmd.hh"
 #include "gz/sim/components/JointPositionReset.hh"
 #include "gz/sim/components/JointType.hh"
+#include "gz/sim/components/JointForce.hh"
 #include "gz/sim/components/JointVelocity.hh"
 #include "gz/sim/components/JointVelocityCmd.hh"
 #include "gz/sim/components/JointVelocityLimitsCmd.hh"
 #include "gz/sim/components/JointVelocityReset.hh"
+#include "gz/sim/components/JointForce.hh"
```
Comment (iche033) [inline src/systems/physics/Physics.cc:124]: CI warning:
``` 
"gz/sim/components/JointForce.hh" already included at /github/workspace/src/systems/physics/Physics.cc:119  
```

**Comment by azeey** [comment]
@Fixit-Davide  we are now in Feature freeze for Gazebo Ionic. Since PR is already open, I've applied the `beta` label which means it's slotted for the Ionic release if we can get it merged before 08/28/2024. Do you think that's feasible? If not, please let me know and I'll remove the `beta` label.

**Comment by azeey** [comment]
To focus our efforts on the Ionic release, I'll go ahead and remove the `beta` label from this. I encourage you to respond to the feedback, but we'll hold off from merging this until Ionic is released.

**Pull request closed without merging**